pkgs/desktops/gnome-3/misc/gnome-flashback/default.nix

```diff
@@ -54,7 +53,11 @@ let
     "org.gnome.SettingsDaemon.Wacom"
     "org.gnome.SettingsDaemon.XSettings"
   ];
-  requiredComponents = wmName: "RequiredComponents=${lib.concatStringsSep ";" ([ wmName ] ++ requiredComponentsCommon ++ requiredComponentsGsd)};";
+  requiredComponents = wmName: enableGnomePanel: "RequiredComponents=${lib.concatStringsSep ";" ([ wmName ] ++ requiredComponentsCommon enableGnomePanel ++ requiredComponentsGsd)};";
+
+  # Make an ASCII filename-friendly name for a wm configuration, similar to Nix.
+  mkWmName = args@{ wmLabel, wmCommand, enableGnomePanel }: builtins.substring 0 32 (builtins.hashString "sha256" (builtins.toJSON args));
```

Comment:
A benefit of human readable name is that it is clearer in the logs.

Reply:
I think this is better for the user, since this name is only needed internally and this way it can't be named in a way that breaks things.

Reply:
Alternately, we might use `slugify wmLabel` as the default value for the name and make it optional. Possibly suffix it with the hash to avoid conflicts.

```nix
slugify =
  let
    allowed = lib.lowerChars ++ lib.upperChars ++ lib.stringToCharacters "1234567890-_";
  in
    lib.stringAsChars (c: if builtins.elem c allowed then c else "_");
```

Reply:
And `types.strMatching "[a-zA-Z0-9_-]+"` for the option would work.

Reply:
I think it's fine how it is now. This identifier is only needed to get GNOME to find it internally, and exposing it would be a leaky abstraction IMO.

Reply:
I've done the slugify approach.

Reply:
Actually, I just remembered that the identifier is needed for auto-login (`services.xserver.displayManager.defaultSession`).

Reply:
Yeah, this is not particularly nice:

```
trace: Default graphical session, 'gnome-flashback-xmonad', not found.
Valid names for 'services.xserver.displayManager.defaultSession' are:
  gnome
  gnome-xorg
  gnome-flashback-Metacity-4175ca004519ae58f559894fbcaadd3f
  gnome-flashback-XMonad-8382231e25a9fde7159467bbc610bdbf
  sm.puri.Phosh
```

Reply:
> BC break

@jtojnar What's "BC"?

Reply:
@nh2 backwards compatibility

Reply:
@jtojnar OK fair enough, I put the `wmName` back in.